### PR TITLE
Add vote command

### PR DESCRIPTION
**Pull request by Bedlessly**
did I do this right?
Its been a while since ive done a pull request on github

### CODE REVIEW

File: src/commands/vote.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  constructor() {
+    super("vote", {
+      description: (language: Language) =>
+        language.get("VOTE_COMMAND_DESCRIPTION"),
+      clientPermissions: ["SEND_MESSAGES"],
+      aliases: ["dbl", "top.gg", "dboats", "discord.boats"],
+    });
+  }
+
+  async exec(message: FireMessage, args: { name: string }) {
```
Comment (GamingGeek) [inline src/commands/vote.ts:15]:
```suggestion
  async exec(message: FireMessage) {
```
There are no args in the command options so the exec method shouldn't take args

File: src/commands/vote.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    await message.channel.send("https://fire-is-the.best/vote");
```
Comment (GamingGeek) [inline src/commands/vote.ts:16]:
```suggestion
    await message.channel.send("https://fire-is-the.best/vote");
```
A constant should probably be used for this URL

File: src/commands/vote.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { FireMessage } from "../../lib/extensions/message";
+import { Language } from "../../lib/util/language";
+import { Command } from "../../lib/util/command";
+
+export default class extends Command {
+  constructor() {
+    super("vote", {
+      description: (language: Language) =>
+        language.get("VOTE_COMMAND_DESCRIPTION"),
```
Comment (GamingGeek) [inline src/commands/vote.ts:9]: You never added this to en-US so it'll just return default

**Review by GamingGeek — CHANGES_REQUESTED** [review]

File: src/commands/vote.ts (hunk 1 above)
Comment (Bedlessly) [reply src/commands/vote.ts:15]: ok, will remember that

File: src/commands/vote.ts (hunk 3 above)
Comment (Bedlessly) [reply src/commands/vote.ts:9]: i couldnt find where the languages were at, sorry

File: src/commands/vote.ts (hunk 2 above)
Comment (Bedlessly) [reply src/commands/vote.ts:16]: as in (const voteUrl: string = "https://fire-is-the.best/vote")

Comment (GamingGeek) [reply src/commands/vote.ts:16]: > as in (const voteUrl: string = "https://fire-is-the.best/vote")

No, see [/lib/util/constants.ts](https://github.com/FireDiscordBot/bot/blob/rewrite/typescript/lib/util/constants.ts)

File: src/commands/vote.ts (hunk 3 above)
Comment (GamingGeek) [reply src/commands/vote.ts:9]: > i couldnt find where the languages were at, sorry

look again then

**Comment by Bedlessly** [comment]
Ok, I added a constant and added the VOTE_COMMAND_DESCRIPTION.
Can it be accepted now?

File: src/languages/en-US.ts [hunk 4]
```diff
@@ -27,6 +27,7 @@ export default class extends Language {
 This will **not** rename existing users`,
         PING_COMMAND_DESCRIPTION: "Shows you my ping to discord's servers",
         PING_INITIAL_MESSAGE: "Pinging...",
+        VOTE_COMMAND_DESCRIPTION: "Vote for Fire to get premium.",
```
Comment (GamingGeek) [inline src/languages/en-US.ts:30]:
```suggestion
        VOTE_COMMAND_DESCRIPTION: "Vote for Fire to get premium.",
```
??????????

**Review by GamingGeek — CHANGES_REQUESTED** [review]

File: src/languages/en-US.ts (hunk 4 above)
Comment (Bedlessly) [reply src/languages/en-US.ts:30]: what did i do wrong now

Comment (GamingGeek) [reply src/languages/en-US.ts:30]: > what did i do wrong now

since when does voting give you premium? I made the damn bot and I didn't know this

Comment (Bedlessly) [reply src/languages/en-US.ts:30]: oh oops, i miss interpreted it

Comment (Bedlessly) [reply src/languages/en-US.ts]: ok fixed

**Comment by Bedlessly** [comment]
👍

File: src/languages/en-US.ts [hunk 5]
```diff
@@ -27,7 +27,7 @@ export default class extends Language {
 This will **not** rename existing users`,
         PING_COMMAND_DESCRIPTION: "Shows you my ping to discord's servers",
         PING_INITIAL_MESSAGE: "Pinging...",
-        VOTE_COMMAND_DESCRIPTION: "Vote for Fire to get premium.",
+        VOTE_COMMAND_DESCRIPTION: "Vote for Fire.",
```
Comment (GamingGeek) [inline src/languages/en-US.ts:30]:
```suggestion
        VOTE_COMMAND_DESCRIPTION: "Vote for Fire.",
```
The description should match the current one on the commands page

Comment (Bedlessly) [reply src/languages/en-US.ts:30]: ok

File: src/commands/vote.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+      clientPermissions: ["SEND_MESSAGES"],
+      aliases: ["dbl", "top.gg", "dboats", "discord.boats"],
+    });
+  }
+
+  async exec(message: FireMessage) {
+    await message.channel.send(constants.url.FireVote);
```
Comment (GamingGeek) [inline src/commands/vote.ts:17]: The command should function the same as the existing Python version which it currently does not

**Review by GamingGeek — CHANGES_REQUESTED** [review]

**Comment by Bedlessly** [comment]
Ok I think im doing what python is doing now.

```py
v = 'vote' if ctx.invoked_with == 'vote' else ''
```
but i dont know if this is important or not

**Comment by Bedlessly** [comment]
im not a python person

File: src/commands/vote.ts [hunk 7]
```diff
@@ -14,6 +14,6 @@ export default class extends Command {
   }
 
   async exec(message: FireMessage) {
-    await message.channel.send(constants.url.FireVote);
+    return await message.channel.send(constants.url.FireVote);
```
Comment (GamingGeek) [inline src/commands/vote.ts:17]:
```suggestion
    return await message.channel.send(constants.url.FireVote);
```
adding return changed nothing about the function of the command

Comment (Bedlessly) [reply src/commands/vote.ts:17]: sorry, guess I shouldnt have tried to help since im loosing brain cells right now and cant think

Comment (Bedlessly) [reply src/commands/vote.ts:17]: ill close the pull request

**Pull request closed without merging**